### PR TITLE
completed game

### PR DESCRIPTION
**Pull request by NathiriRuby**
Had a little bit of a panic when I realized I had been working off of the master branch the whole time ._.
But it's okay now.
Definitely works.
Definitely.
Probably.
But Definitely.
I'm gonna take a nap.

### CODE REVIEW

**Comment by NathiriRuby** [comment]
43810316
Christian Ventura

**Comment by pragdave** [comment]
:)

File: hangman/lib/hangman.ex [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  end
+
+  # def auto_win(game) do           #debug
+  #   %{game | game_state: :won}
+  # end
+
+  # def auto_lose(game) do          #debug
+  #    %{game | game_state: :lose}
+  # end
+
```
Comment (pragdave) [inline hangman/lib/hangman.ex:69]: I count 30 lines in this function, which makes it about 27 lines too long :)

I also see conditional logic nested three levels deep, which is two more levels than I like.

See the overall comment I made at the top level.

**Comment by pragdave** [comment]
I have my own tests that I run against everyone's code. Unfortunately, this code doesn't pass them, as you aren't following the required format for the tally. 

* you're returning strings and not values for the status, letters, and turns left
* you're returning the charlist `'_'` rather than the string `"_"` for unknown letters
* you return `:ok` if a letter is already used (because you have an IO.puts there)

Not having tests costs 5 points, and not passing mine drops 25 (out of the 70 points allotted to "the code works").

You haven't put the implementation in a subdirectory and delegated to it from the top.

The `make_move` function is decidedly not functional in style, and is way too big.

So, overall, right now you're looking at a scoring 70 out of 115.

But that's not how you will learn to do it differently, so here's my suggestion. If you want to recode the main module and add some tests, I'll grade this afresh, with a penalty of 20 out of 115 (so your baseline grade would be 95/115).

Totally your call. If you want to do this, come and see me before or after class and I'll talk you through some areas where you could score better.

**Pull request closed without merging**